baseplate/observers/prometheus.py

```diff
@@ -109,13 +113,84 @@ def on_child_span_created(self, span: Span) -> None:
         else:
             observer = PrometheusClientSpanObserver()
 
-        observer.on_set_tag("protocol", self.protocol)
         span.register(observer)
 
 
 class PrometheusClientSpanObserver(SpanObserver):
     def __init__(self) -> None:
-        logger.debug("PrometheusClientSpanObserver not implemented")
+        self.tags: Dict[str, Any] = {}
+        self.start_time: Optional[int] = None
+        self.metrics: Optional[
+            Union[PrometheusHTTPClientMetrics]
+        ] = None  # Add PrometheusThriftClientMetrics when implemented
+
+    def on_set_tag(self, key: str, value: Any) -> None:
+        self.tags[key] = value
+
+    def get_prefix(self) -> str:
+        return f"{self.protocol}_client"
+
+    def get_labels(self) -> Dict[str, str]:
+        return self.tags
+
+    @property
+    def protocol(self) -> str:
+        return self.tags.get("protocol", "unknown")
+
+    def set_metrics_by_protocol(self) -> None:
+        if self.protocol == "http":
+            self.metrics = PrometheusHTTPClientMetrics()
+        elif self.protocol == "thrift":
+            # self.metrics = PrometheusThriftClientMetrics()
+            logger.warning("PrometheusThriftClientMetrics not implemented.")
```

Comment:
Will this spam thift application logs when prometheus metrics aren't setup yet? How often will this message be logged?

Reply:
Yes technically but we will merge everything together and cut a release from there. So I don't believe this should be a problem.

Reply:
Updated them to debug since I was in there